### PR TITLE
[ci] Addon Test Harness

### PR DESCRIPTION
**Pull request by shaneutt**
This adds CI tests for addons. In this iteration for scope and speed we're covering just the Zookeeper addon, later iterations will cover the suite.

### CODE REVIEW

File: test/addons_test.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cluster.Cleanup()
+
+	if err := temp.DeployController(cluster); err != nil {
+		t.Fatal(err)
+	}
+
+	b, err := ioutil.ReadFile("../addons/zookeeper/0.x/zookeeper.yaml")
```
Comment (shaneutt) [inline test/addons_test.go]: As per the description, this is just step one, another PR will change coverage to ALL addons but there's some Catalog functionality that is not ready yet for this.

**Comment by joejulian** [comment]
https://teamcity.mesosphere.io/viewLog.html?tab=buildLog&logTab=tree&filter=debug&expand=all&buildId=2448613&_focus=440

No artifacts. Is this expected?

**Comment by shaneutt** [comment]
> https://teamcity.mesosphere.io/viewLog.html?tab=buildLog&logTab=tree&filter=debug&expand=all&buildId=2448613&_focus=440
> 
> No artifacts. Is this expected?

There's practically nothing

> https://teamcity.mesosphere.io/viewLog.html?tab=buildLog&logTab=tree&filter=debug&expand=all&buildId=2448613&_focus=440
> 
> No artifacts. Is this expected?

I'm not certain, but also this performs its job entirely in a container does that have any effect on TC collecting artifacts?

**Comment by jkoelker** [comment]
> https://teamcity.mesosphere.io/viewLog.html?tab=buildLog&logTab=tree&filter=debug&expand=all&buildId=2448613&_focus=440
> 
> No artifacts. Is this expected?

I believe so, the config in teamcity is just to run the go tests, but the larger question of why have it list them as artifacts remains.

**Comment by joejulian** [comment]
I guess my question is more about outputting artifacts. We used to have a way to dump logs and objects from kind. Did we lose that?

**Comment by joejulian** [comment]
(used to in kubeaddons, that is)

**Comment by shaneutt** [comment]
It should be practically nothing, all logging currently is done to stdout.

**Comment by shaneutt** [comment]
No artifacts are expected to be collected at this time, the ones that failed to collect are just copied over configuration for the TC build for this repo. I've since removed that configuration and we should be all set. :+1:

**Pull request merged**